### PR TITLE
Np 1384 update project config

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,1663 @@
+[*]
```
Comment (axthosarouris) [inline .editorconfig:1]: copied from java-codestyle

File: config/checkstyle/checkstyle-author.xsl [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
-  <xsl:output indent="yes" method="html"/>
   <xsl:decimal-format decimal-separator="." grouping-separator=","/>
+  <xsl:key match="//file/error[contains(@message,'@author')]" name="keyAuthorID" use="@message"/>
```
Comment (axthosarouris) [inline config/checkstyle/checkstyle-author.xsl:3]: copied from java-codestlye

File: config/checkstyle/checkstyle-frames.xsl [hunk 3]
```diff
@@ -57,8 +57,8 @@
    <http://www.apache.org/>.
    -->
 
-  <xsl:output encoding="US-ASCII" indent="yes" method="html"/>
   <xsl:decimal-format decimal-separator="." grouping-separator=","/>
+  <xsl:output encoding="US-ASCII" indent="yes" method="html"/>
```
Comment (axthosarouris) [inline config/checkstyle/checkstyle-frames.xsl:61]: copied from java-codestyle project

File: config/checkstyle/checkstyle-simple.xsl [hunk 4]
```diff
@@ -4,13 +4,13 @@
 
   <xsl:template match="/">
     <html>
-      <head>
-        <title>Sun Coding Style Violations</title>
-      </head>
       <body bgcolor="#FFFFEF">
+        <hr align="left" size="1" width="95%"/>
```
Comment (axthosarouris) [inline config/checkstyle/checkstyle-simple.xsl:8]: copied from java-codestyle project

File: config/checkstyle/checkstyle-text.xsl [hunk 5]
```diff
@@ -6,27 +6,28 @@
     Coding Style Check Results
     --------------------------
     Total files checked:
-    <xsl:number level="any" value="count(descendant::file)"/>
+    <xsl:apply-templates/>
```
Comment (axthosarouris) [inline config/checkstyle/checkstyle-text.xsl:9]: copied from java-codestyle project

File: config/checkstyle/checkstyle.xml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <!DOCTYPE module PUBLIC
-        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
-        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
```
Comment (axthosarouris) [inline config/checkstyle/checkstyle.xml:3]: copied from java-codestyle project

File: config/pmd/ruleset.xml [hunk 7]
```diff
@@ -19,7 +19,7 @@
 
   <rule ref="category/java/bestpractices.xml">
     <exclude name="SystemPrintln"/>
-	<exclude name="GuardLogStatement"/>
+    <exclude name="GuardLogStatement"/>
```
Comment (axthosarouris) [inline config/pmd/ruleset.xml:22]: copied from java-codestyle project

File: create-doi-request/build.gradle [hunk 8]
```diff
@@ -0,0 +1,3 @@
+dependencies {
+    implementation project(":doi-requests-commons")
```
Comment (axthosarouris) [inline create-doi-request/build.gradle:2]: Here we will build the new handlers for creating and updating  a doi-request

File: settings.gradle [hunk 9]
```diff
@@ -1,3 +1,4 @@
 rootProject.name = 'nva-doi-requests-api'
 include 'doi-requests-commons'
 include 'find-doi-requests'
+include 'create-doi-request'
```
Comment (axthosarouris) [inline settings.gradle:4]: create sub-project

File: config/pmd/ruleset.xml [hunk 10]
```diff
@@ -61,6 +60,13 @@
 
   </rule>
 
+  <rule ref="category/java/codestyle.xml/ShortMethodName">
```
Comment (axthosarouris) [inline config/pmd/ruleset.xml:63]: Allow methods with name "of"

**Review by janbj — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**